src/object/obj_tx.c

```diff
@@ -1814,25 +1811,22 @@ dc_tx_commit_trigger(tse_task_t *task, struct dc_tx *tx, daos_tx_commit_t *args)
 	D_MUTEX_UNLOCK(&tx->tx_lock);
 
 	rc = daos_rpc_send(req, task);
-	if (rc != 0)
+	if (rc != 0) {
+		D_MUTEX_LOCK(&tx->tx_lock);
```

Comment:
Where is this lock dropped?  This look wrong to me.

Reply:
This function is called from dc_tx_commit() only and expect this lock is acquired after return. So, I have to stay with this logic. I'm also not happy from this implementation.

Reply:
And it's OK to drop it and reclaim it?  This might mean that status has changed compared to keeping the lock held throughout.

Reply:
@Nasf-Fan, could you clarify the logic here please? How this code can be elaborated?